gobblin-yarn/src/main/java/org/apache/gobblin/yarn/YarnService.java

```diff
@@ -473,6 +466,8 @@ public synchronized void requestTargetNumberOfContainers(YarnContainerRequestBun
         ContainerInfo containerInfo = entry.getValue();
         if (!inUseInstances.contains(containerInfo.getHelixParticipantId())) {
           containersToRelease.add(containerInfo.getContainer());
+          requestedContainerCountMap.put(containerInfo.getHelixTag(),
+              requestedContainerCountMap.get(containerInfo.getHelixTag()) - 1);
```

Comment:
Here should be the place to decrease requestedContainerCountMap, as we're purposely releasing containers and won't spawn replacements.